### PR TITLE
Custom apis

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: tests/test_custom.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from indicoio.custom import Collection, collections
+import time
+
+collection_name = "__test_python_text__"
+test_data = [['input 1', 'label 1'], ['input 2', 'label 2'], ['input 3', 'label 3'], ['input 4', 'label 4']]
+
+image_collection_name = "__test_python_image__"
+image_test_data = [
+    ["https://i.imgur.com/xUX1rvY.png", "dog"],
+    ["https://i.imgur.com/xUX1rvY.png", "dog"],
```
Comment (sihrc) [inline tests/test_custom.py]: these two are the same image urls

File: tests/test_custom.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            Collection(image_collection_name).clear()
+        except IndicoError:
+            pass
+
+    def test_add_predict(self):
+        collection = Collection(collection_name)
+        collection.add_data(test_data)
+        collection.train()
+        collection.wait()
+        result = collection.predict(test_data[0][0])
```
Comment (sihrc) [inline tests/test_custom.py]: Is this right? I get a dictionary of scores

File: tests/test_custom.py (hunk 1 above)
Comment (madisonmay) [inline tests/test_custom.py]: it doesn't actually matter for the purpose of the test.

File: tests/test_custom.py (hunk 2 above)
Comment (madisonmay) [inline tests/test_custom.py]: That's what you should get?

Comment (sihrc) [inline tests/test_custom.py]: :+1: i don't remember what i was thinking before. so cool.

File: indicoio/custom/custom.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+        .. code-block:: python
+
+           >>> text = "London Underground's boss Mike Brown warned that the strike ..."
+           >>> indicoio.add_data([[text, .5]])
+        """
+        batch = isinstance(data[0], list)
+        if batch:
+            X, Y = zip(*data)
+            X = image_preprocess(X, batch=batch)
```
Comment (anneCarlson) [inline indicoio/custom/custom.py]: Why are we doing image processing on text?

**Pull request merged**